BlueprintUI/Sources/Layout/Builder/StackChild.swift

```diff
@@ -0,0 +1,87 @@
+import UIKit
+
+/// `StackChild` is a wrapper which allows an element to define its StackLayout traits and Keys.
+/// This struct is particularly useful when working with `@StackElementBuilder`.
+/// By default, elements will default to a nil key and the default `StackLayout.Traits` initializer.
+/// `@StackElementBuilder` will check every child to see if it can be type cast to a `StackChild`
+/// and then pull of the given traits and key and then apply those to the stack
+public struct StackChild: ProxyElement {
```

Comment:
It feels weird to me that this is an element. I would suggest inverting these, making `StackChild` the currency of the builder, and handling plain elements with a `buildExpression` func.

Something like:

```swift
struct StackChild {
    var element: Element
    var traits: Traits = .defaultTraits
}

@_functionBuilder
struct StackBuilder {
    typealias Component = [StackChild]

    static func buildBlock(_ children: Component...) -> Component {
        children.flatMap { $0 }
    }

    static func buildExpression(_ element: Element) -> Component {
        // promote naked element to a child with default traits
        [StackChild(element: element)]
    }
}

extension Element {
    // or perhaps separate modifiers for priority, key, etc
    func stackTraits(
        growPriority: CGFloat = 1,
        shrinkPriority: CGFloat = 1,
        key: AnyHashable? = nil
    ) -> StackChild {
        StackChild(
            element: self,
            traits: Traits(
                growPriority: growPriority,
                shrinkPriority: shrinkPriority,
                key: key
            )
        )
    }
}
```

Reply:
I do like the idea of changing the currency of the builder...but one downside of changing to be a StackChild is that we will now need new builders for different container based views such as EqualStacks and GridRows. What seems odd about encapsulating stack information in an Element? It seems like many elements in the library simply carry along extra information to handle layout, styling, interaction, etc.

Reply:
> one downside of changing to be a StackChild is that we will now need new builders for different container based views such as EqualStacks and GridRows

Yeah, but that seems like the right approach to me. Each of those containers has a different child type, with different traits. With the approach you have now, each container has to do some version of an `element as? StackChild` downcast to get the type it wants, and that goes away if we use the right currency for each container.

> It seems like many elements in the library simply carry along extra information to handle layout, styling, interaction, etc.

Elements carry this information but do not share it; they are generally self-contained and loosely coupled. They work together by composition, while remaining opaque to each other.

Reply:
This makes sense to me. Thanks for explaining! I will go ahead and make these changes 👍

Reply:
@watt I created two new builders that use StackChild and GridRowChild as a currency (see latest commit). I definitely like not having to cast Elements to StackChild anymore!